File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for val in tokens:
            if val != '+' and val != '-' and val != '*' and val != '/':
                stack.append(int(val))
            else:
                num1 = stack.pop()
                num2 = stack.pop()
                print(num1, "", val, "", num2)
                if val == '+':
                    stack.append(int(num2 + num1))
                elif val == '-':
                    stack.append(int(num2 - num1))
                elif val == '*':
                    stack.append(int(num2 * num1))
                elif val == '/':
                    stack.append(int(num2 / num1))
        return stack[0]

File: Data Structures & Algorithms/baseball-game/submission-2.go
func calPoints(operations []string) int {
	var stack []int
	var total int
	for _, operation := range operations{
		if operation == "+"{
			n := len(stack)
			num1 := stack[n-1]
			num2 := stack[n-2]
			stack = append(stack, num1+num2)
			total += (num1+num2)
		}else if operation == "C"{
			total -= stack[len(stack)-1]
            stack = stack[:len(stack)-1]
		}else if operation == "D"{
			num := stack[len(stack)-1]
			stack = append(stack, 2*num)
			total += 2*num
		}else{
			num, _ := strconv.Atoi(operation)
            stack = append(stack, num)
            total += num
		}
	}
	return total
}

func Pop(stack []int) int{
	topIndex := len(stack) - 1
	topElement := stack[topIndex]
	stack = stack[:topIndex]
	return topElement
}